### PR TITLE
Topic/warnings 2.10.x

### PR DESCRIPTION
**Pull request by paulp**
Eliminated most non-deprecation warnings in trunk.

I didn't know if this would go in 2.10.x so I'm opening separate pull requests.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -1294,7 +1294,7 @@ QUICK BUILD (QUICK)
     <scalacfork
       destdir="${build-quick.dir}/classes/scalacheck"
       compilerpathref="quick.classpath"
-      params="${scalac.args.quick}"
+      params="${scalac.args.quick} -nowarn"
```
Comment (adriaanm) [inline build.xml:1297]: I was thinking of going the other way: -Xfatal-warnings

**Comment by adriaanm** [comment]
LGTM. checkCheckable is much nicer now. I think this should go into RC1.

File: build.xml (hunk 1 above)
Comment (paulp) [inline build.xml:1297]: The problem is that we don't control scalacheck and I'm not super keen on maintaining local changes just to pacify the warning monster.

Comment (adriaanm) [inline build.xml:1297]: Oh, sorry. I missed that it was in scalacheck.

**Pull request merged**